src/main/java/de/unijena/cheminf/mortar/model/fragmentation/algorithm/AlkylStructureFragmenter.java

```diff
@@ -0,0 +1,710 @@
+                        tmpFragmentSet.addAtomContainer(tmpContainer);
+                    }
+                } else {
+                    tmpFragmentSet.addAtomContainer(anAtomContainer);
+                }
+            }
+            return tmpFragmentSet;
+        } catch (Exception anException) {
+            AlkylStructureFragmenter.this.logger.log(Level.WARNING, anException + " Connectivity Checking failed at molecule: " + anAtomContainer.getID(), anException);
+            throw new IllegalArgumentException("An Error occurred during Connectivity Checking: " + anException.toString() +
+                    ": " + anAtomContainer.getProperty(Importer.MOLECULE_NAME_PROPERTY_KEY));
+        }
+    }
+
+    /**
+     * Private method to saturate a given molecule with implicit hydrogens after fragmentation.
+     *
+     * @param anUnsaturatedACSet IAtomContainerSet whose atomcontainers are to be saturated
+     * @return List of processed atomcontainers, @null if given Set is empty
+     * @throws CDKException if CDKHydrogenAdder throws an exception
+     */
+    private List<IAtomContainer> saturateWithImplicitHydrogen(IAtomContainerSet anUnsaturatedACSet) throws CDKException {
+        Objects.requireNonNull(anUnsaturatedACSet, "Given IAtomContainerSet is null.");
+        try {
+            List<IAtomContainer> tmpSaturatedFragments = new ArrayList<>(anUnsaturatedACSet.getAtomContainerCount());
+            if (!anUnsaturatedACSet.isEmpty() && !anUnsaturatedACSet.getAtomContainer(0).isEmpty()) {
+                CDKHydrogenAdder tmpAdder = CDKHydrogenAdder.getInstance(anUnsaturatedACSet.getAtomContainer(0).getBuilder());
+                for (IAtomContainer tmpAtomContainer: anUnsaturatedACSet.atomContainers()) {
+                    if (tmpAtomContainer != null && !tmpAtomContainer.isEmpty()) {
+                        AtomContainerManipulator.percieveAtomTypesAndConfigureAtoms(tmpAtomContainer);
+                        if (this.fragmentSaturationSetting.get().equals(FragmentSaturationOption.HYDROGEN_SATURATION.name())) {
+                            tmpAdder.addImplicitHydrogens(tmpAtomContainer);
+                            tmpSaturatedFragments.add(tmpAtomContainer);
+                        } else {
+                            tmpSaturatedFragments.add(tmpAtomContainer);
+                        }
+                    }
+                }
+            }
+
+            //this.fragmentationEnd = System.nanoTime();
+            //System.out.println(this.fragmentationEnd - this.fragmentationStart);
+
+            return tmpSaturatedFragments;
+        } catch (CDKException anException) {
+            AlkylStructureFragmenter.this.logger.log(Level.WARNING, anException
+                + " Unable to add Implicit Hydrogen");
+            throw new CDKException("Unexpected error occurred during implicit hydrogen adding at " +
+                "hydrogen saturation of molecule, " + anException.toString(), anException);
```

Comment:
does this really add sth important to the already existing exception? If not, I would not catch and re-throw it here.